src/main/java/com/academia/andruhovich/library/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,209 @@
+        return new OrderContentWrapper(bookDto, quantity);
+    }
+
+    protected BigDecimal calculateTotalPrice(Map<Long, Integer> orderContent, List<Book> books) {
+        return books.stream()
+                .map(book -> getBigDecimal(orderContent, book))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+    }
+
+    private BigDecimal getBigDecimal(Map<Long, Integer> orderContent, Book book) {
+        BigDecimal quantity = BigDecimal.valueOf(orderContent.get(book.getId()));
+        return book.getPrice().multiply(quantity);
+    }
+
+    protected BigDecimal calculateTotalPrice(Set<OrderContentWrapper> orderContent, List<Book> books) {   //refactoring
+        return books.stream()
+                .map(book -> getBookSetBigDecimalPrice(orderContent, book))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+    }
+
+    private BigDecimal getBookSetBigDecimalPrice(Set<OrderContentWrapper> orderContent, Book book) {
+        OrderContentWrapper currentBookAndQuantity = orderContent.stream()
+                .filter(element -> book.getId().equals(element.getBook().getId()))
+                .findFirst()
+                .orElseThrow(() ->
+                        new ResourceNotFoundException(String.format(BOOK_NOT_FOUND, book.getId())));
+        Integer quantity = currentBookAndQuantity.getQuantity();
+        BigDecimal quantityBigDecimal = BigDecimal.valueOf(quantity);
+        return book.getPrice().multiply(quantityBigDecimal);
+    }
+
+    protected Order recalculatePriceAndHistory(Order order) {
+        if (order.getHistory().startsWith("\"")) {
+            order.setHistory(order.getHistory().replace("\\", ""));
+            order.setHistory(order.getHistory().substring(1, order.getHistory().length() - 1));
+        }
+
+        Set<OrderContentWrapper> orderContentWrappers = jsonConverter.jsonStringToCollection(order.getHistory());
+        List<Long> booksIds = orderContentWrappers.stream()
+                .map(OrderContentWrapper::getBook)
+                .map(BookDto::getId)
+                .collect(Collectors.toList());
+        List<Book> synchronisedBooks = bookRepository.findAllById(booksIds);
+
+        Set<OrderContentWrapper> synchronisedOrderContentWrappers = synchronisedBooks.stream()
+                .map(Book::getId)
+                .map(id -> orderContentWrappers.stream()
+                        .filter(orderContentWrapper -> id.equals(orderContentWrapper.getBook().getId()))
+                        .findFirst()
+                        .orElseThrow(() ->
+                                new ResourceNotFoundException(String.format(BOOK_NOT_FOUND, id)))
+                )
+                .collect(Collectors.toSet());
+
+        order.setHistory(jsonConverter.collectionToJsonString(synchronisedOrderContentWrappers));
+        order.setTotalPrice(calculateTotalPrice(synchronisedOrderContentWrappers, synchronisedBooks));
+
+        return orderRepository.save(order);
```

Comment:
move orderRepository.save(order); out of this method return  just return order

Reply:
Refactored it